pkg/kf/commands/service-bindings/integration_test.go

```diff
@@ -0,0 +1,166 @@
+			withServiceInstance(ctx, kf, func(ctx context.Context) {
+				withServiceBinding(ctx, t, kf, func(ctx context.Context) {
+					// Restart so that env vars are injected from the secret into app
+					kf.Restart(ctx, AppFromContext(ctx))
+					vcapServicesOutput := kf.VcapServices(ctx, AppFromContext(ctx))
+					AssertContainsAll(t, strings.Join(vcapServicesOutput, "\n"), []string{AppFromContext(ctx),
+						ServiceInstanceFromContext(ctx), creds})
+				})
+			})
+		})
+	})
+}
+
+var checkOnce sync.Once
+
+func checkClusterStatus(t *testing.T) {
+	checkOnce.Do(func() {
+		testIntegration_Doctor(t)
+	})
+}
+
+// testIntegration_Doctor runs the doctor command. It ensures the cluster the
+// tests are running against is in good shape.
+func testIntegration_Doctor(t *testing.T) {
+	RunKfTest(t, func(ctx context.Context, t *testing.T, kf *Kf) {
+		kf.Doctor(ctx)
+	})
+}
+
+func withServiceBroker(ctx context.Context, t *testing.T, kf *Kf, callback func(newCtx context.Context)) {
+	brokerAppName := fmt.Sprintf("integration-broker-app-%d", time.Now().UnixNano())
+	brokerName := "fake-broker"
+
+	// Push a mock service broker app and then clean it up.
+	kf.Push(ctx, brokerAppName,
+		"--path", filepath.Join(RootDir(ctx, t), "./samples/apps/service-broker"),
+	)
+
+	defer kf.Delete(ctx, brokerAppName)
+
+	// Register the mock service broker to service catalog, and then clean it up.
+	kf.AddServiceBroker(ctx, brokerName, internalBrokerUrl(brokerAppName, SpaceFromContext(ctx)))
+
+	// Temporary solution to allow service broker registration to complete.
+	// TODO: Add flag to run the command synchronously.
+	time.Sleep(2 * time.Second)
+	defer kf.DeleteServiceBroker(ctx, brokerName)
+
+	ctx = ContextWithBroker(ctx, brokerName)
+	callback(ctx)
+}
+
+func withServiceInstance(ctx context.Context, kf *Kf, callback func(newCtx context.Context)) {
+	serviceClass := "fake-service" // service class provided by the mock broker
+	servicePlan := "fake-plan"     // service plan provided by the mock broker
+	serviceInstanceName := "int-service-instance"
+
+	kf.CreateService(ctx, serviceClass, servicePlan, serviceInstanceName)
+
+	// Temporary solution to allow service instance creation to complete.
+	// TODO: Add flag to run the command synchronously.
+	time.Sleep(2 * time.Second)
+
+	defer kf.DeleteService(ctx, serviceInstanceName)
+
+	ctx = ContextWithServiceClass(ctx, serviceClass)
+	ctx = ContextWithServicePlan(ctx, servicePlan)
+	ctx = ContextWithServiceInstance(ctx, serviceInstanceName)
+	callback(ctx)
+}
+
+func withServiceBinding(ctx context.Context, t *testing.T, kf *Kf, callback func(newCtx context.Context)) {
+	appName := fmt.Sprintf("integration-binding-app-%d", time.Now().UnixNano())
+
+	// Push the envs app, which will have a binding to a service instance, and then clean it up.
+	kf.Push(ctx, appName,
+		"--path", filepath.Join(RootDir(ctx, t), "./samples/apps/envs"),
+	)
+	defer kf.Delete(ctx, appName)
```

Comment:
I think this could be split into `withApp` which would push the given app and `withServiceBinding` which would bind to a given app. Then `withServiceBroker` could re-use `withApp` to push the fake broker

Reply:
done